src/models/trainerMatchingDao.js

```diff
@@ -0,0 +1,190 @@
+        p.content AS content,
+        p.created_at AS createdAt ,
+        u.gender AS gender,
+        t.specialized AS specialized,
+        u.img_url AS imgUrl
+        FROM products p 
+        JOIN trainers t ON t.id = p.trainer_id
+        JOIN users u ON u.id = t.user_id
+        WHERE p.status = 1
+        ${categoryQuery}
+        ${genderQuery}
+        ${trainerCheckQuery}
+        ${sortQuery}
+        ${products};
+        `);
+  return result;
+};
+
+//트레이너 상세 정보
+const getTrainerMatchingDetail = async (userId, productsId) => {
+  const [result] = await AppDataSource.query(
+    `
+        SELECT 
+        p.id AS id,
+        p.trainer_id AS trainerId,
+        p.available_area AS availableArea,
+        p.available_time AS availableTime,
+        p.category_name AS categoryName,
+        p.term AS term,
+        p.price AS price,
+        p.content AS content,
+        p.created_at AS createdAt ,
+        u.gender AS gender,
+        t.specialized AS specialized,
+        u.img_url AS imgUrl
+        FROM products p 
+        JOIN trainers t ON t.id = p.trainer_id
+        JOIN users u ON u.id = t.user_id
+        WHERE p.status = 1
+        AND u.id = ?
+        AND p.id = ?;
+        `,
+    [userId, productsId]
+  );
+  return result;
+};
+
+const isTrainer = async (userId) => {
+  const [result] = await AppDataSource.query(
+    `SELECT CASE
+        WHEN (SELECT u.nickname FROM trainers t
+            JOIN users u ON t.user_id  = u.id
+            WHERE u.id = ?) IS NULL
+            THEN 'false'
+            ELSE 'true'
+        END AS isTrainer;
+    `,
+    [userId]
+  );
+  return result.isTrainer;
+};
+// 구독자 여부 확인
+const isSubscribed = async (userId) => {
+  const [result] = await AppDataSource.query(
+    `SELECT CASE
+        WHEN (SELECT u.nickname FROM sub_orders so
+            JOIN users u ON so.user_id  = u.id
+            WHERE u.id = ?) IS NULL
+            THEN 'false'
+            ELSE 'true'
+        END AS isSubscribed;
+      `,
+    [userId]
+  );
+  return result.isSubscribed;
+};
+const isPostedTrainer = async (productsId, trainerId) => {
+  const [result] = await AppDataSource.query(
+    `SELECT CASE
+        WHEN (SELECT p.id FROM p
+            WHERE p.id = ?
+            AND p.trainer_id = ?) IS NULL
+            THEN 'false'
+            ELSE 'true'
+        END AS isPostedTrainer;
+          `,
+    [productsId, trainerId]
+  );
+  return result.isPostedTrainer;
+};
+
+const findTrainerId = async (userId) => {
+  const [trainerId] = await AppDataSource.query(
+    `
+    SELECT t.id AS id
+    FROM trainer t
+    JOIN users u ON u.id = t.user_id
+    WHERE u.id = ?;`,
+    [userId]
+  );
+  return trainerId.id;
+};
+const findCategoryName = async (trainerId) => {
+  const [category] = await AppDataSource.query(
+    `
+    SELECT wc.category AS name
+    FROM workout_category wc
+    JOIN trainers t ON t.specialized = wc.id
+    WHERE t.id = ?;`,
+    [trainerId]
+  );
+  return category.name;
```

Comment:
```suggestion
const findSpecializedCategoryByTrainerId = async (trainerId) => {
  const [category] = await AppDataSource.query(
    `
    SELECT wc.category AS name
    FROM workout_category wc
    JOIN trainers t ON t.specialized = wc.id
    WHERE t.id = ?;`,
    [trainerId]
  );
  return category.name;
```
어떤 아이디를 통해서 조회하는지가 명시되어 있지 않으면, 해당 함수를 사용하는 곳에서, 응당 categoryId로 오인하고 입력할 가능성이 있습니다. 확인부탁드립니다 :)